src/modules/SdnDiag.Utilities.psm1

```diff
@@ -995,13 +992,33 @@ function Export-ObjectToFile {
         }
     }
     process {
-        # if the object is null, then we will not add it to the array list
-        # this is to prevent the array list from being empty and causing issues with the export
-        if ($null -eq $Object) {
-            return
-        }
+        foreach ($obj in $Object) {
+            if ($null -eq $obj) {
+                continue  # skips this object, continues with next object
+            }
+
+            # we need to enum the object type as there may be lots of extra properties that are not needed
+            # which we will want to remove from the object beofre exporting it out
+            if ($FileType -ieq 'json') {
+                $objectTypeName = $obj.GetType().Name
+                switch -Wildcard ($objectTypeName) {
+                    'CimInstance' {
+                        $obj = $obj | Remove-PropertiesFromObject -PropertiesToRemove 'CimClass','CimInstanceProperties','CimSystemProperties'
+                        break
+                    }
+                    'VMNetworkAdapter*' {
+                        $obj = $obj | Remove-PropertiesFromObject -PropertiesToRemove 'ParentAdapter','CimSession'
+                        break
+                    }
```

Comment:
minor: why bother, let them be there so that one could query the class name and look up the strcuture .

Reply:
@sbgms mainly removing because of the overhead. Passing all the CimProperties results in the operation taking greater than 10 minutes to output the file. The commands for VMNetworkAdapter* are not very efficient in that regard.